lib/rucio/core/replica.py

```diff
@@ -2508,115 +2694,142 @@ def get_and_lock_file_replicas_for_dataset(scope, name, nowait=False, restrict_r
     files, replicas = {}, {}
     if session.bind.dialect.name == 'postgresql':
         # Get content
-        content_query = session.query(models.DataIdentifierAssociation.child_scope,
-                                      models.DataIdentifierAssociation.child_name,
-                                      models.DataIdentifierAssociation.bytes,
-                                      models.DataIdentifierAssociation.md5,
-                                      models.DataIdentifierAssociation.adler32).\
-            with_hint(models.DataIdentifierAssociation,
-                      "INDEX_RS_ASC(CONTENTS CONTENTS_PK) NO_INDEX_FFS(CONTENTS CONTENTS_PK)",
-                      'oracle').\
-            filter(models.DataIdentifierAssociation.scope == scope,
-                   models.DataIdentifierAssociation.name == name)
+        stmt = select(
+            models.DataIdentifierAssociation.child_scope,
+            models.DataIdentifierAssociation.child_name,
+            models.DataIdentifierAssociation.bytes,
+            models.DataIdentifierAssociation.md5,
+            models.DataIdentifierAssociation.adler32,
+        ).with_hint(
+            models.DataIdentifierAssociation,
+            'INDEX_RS_ASC(CONTENTS CONTENTS_PK) NO_INDEX_FFS(CONTENTS CONTENTS_PK)', 'oracle'
```

Comment:
Should `oracle` be on a different line?

Reply:
(Same as: https://github.com/rucio/rucio/pull/6591#discussion_r1668793992)

Changed to:
```
        ).with_hint(
            models.DataIdentifierAssociation,
            'INDEX_RS_ASC(CONTENTS CONTENTS_PK) NO_INDEX_FFS(CONTENTS CONTENTS_PK)',
            'oracle'
        ).outerjoin(
```